src/gsi/hybrid_ensemble_isotropic.F90

```diff
@@ -3806,17 +3866,23 @@ subroutine ckgcov_a_en_new_factorization(ig,z,a_en)
 ! Passed Variables
   integer(i_kind),intent(in   ) :: ig
   type(gsi_bundle),intent(inout) :: a_en(n_ens)
-  real(r_kind),dimension(nval_lenz_en),intent(in   ) :: z
+  real(r_kind),dimension(nval_loc_en),intent(in   ) :: z
+!NOTE:
+! nval_loc_en is the number of horizontally-filtered variables in the domain of each processor.
+! In MGBF-based localization, it is horizontally-local and vertically-global as
+!   nval_loc_en = nhoriz * obj_mgbf(ig)%km_all
+!      and nhoriz = ( obj_mgbf(ig)%im + obj_mgbf(ig)%hx*2 ) * ( obj_mgbf(ig)%jm + obj_mgbf(ig)%hy*2 )
+! In recursive/spectral filter-based localization, it is horizontally-global and vertically-local as
+!   nval_loc_en = nhoriz * ( grd_loc%kend_alloc - grd_loc%kbegin_loc + 1 )
+!      and nhoriz =  grd_loc%nlat     *  grd_loc%nlon     (for regional recursive filter)
+!          nhoriz = ( sp_loc%jcap+1 ) * ( sp_loc%jcap+2 ) (for global spectral filter)
+! but internal array hwork always has
+! dimension grd_loc%nlat * grd_loc%nlon * ( grd_loc%kend_alloc - grd_loc%kbegin_loc + 1 )
+! which just happens to match up with nval_loc_en only for regional recursive filter.
```

Comment:
@shoyokota , I would suggest this line as " which would be used as nval_loc_en when the recursive filter  is used"

Reply:
Corrected.